Compile Error "patterns aren't allowed in methods without bodies"

on a reasonably clean ubuntu 17.10 installation i'm getting the following compile error, confirmed on 3 different but similar machines:

```
   Compiling pkg-config v0.3.8
   Compiling bitflags v0.2.1
   Compiling rustc-serialize v0.3.19
   Compiling semver v0.1.20
   Compiling winapi v0.2.8
   Compiling memalloc v0.1.0
   Compiling termbox-sys v0.2.9
   Compiling bitflags v0.3.3
   Compiling winapi-build v0.1.1
   Compiling num-traits v0.1.35
error[E0642]: patterns aren't allowed in methods without bodies
   --> /home/gtz/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-serialize-0.3.19/src/serialize.rs:147:45
    |
147 |                                             &f_name: &str,
    |                                             ^^^^^^^

   Compiling libc v0.2.15
   Compiling kernel32-sys v0.2.2
   Compiling rand v0.3.14
   Compiling rustc_version v0.1.7
   Compiling alsa-sys v0.0.8
   Compiling num-integer v0.1.32
error: failed to run custom build command for `alsa-sys v0.0.8`
process didn't exit successfully: `/home/gtz/Downloads/microbrust/microbrust/target/debug/build/alsa-sys-8585bd72a32905b6/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "Failed to run `\"pkg-config\" \"--libs\" \"--cflags\" \"alsa\"`: No such file or directory (os error 2)"', src/libcore/result.rs:916:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: aborting due to previous error

error: Could not compile `rustc-serialize`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

On Tue, 2018-06-05 at 09:16 +0000, Jakub Matraszek wrote:
> Thanks from opening an issue. This shouldn't be too hard to fix,
> expect a new release in a few days.

maybe noteworthy: on ubuntu 18.04 i get the same compile error at a
different liocation:

   Compiling winapi-build v0.1.1
error[E0642]: patterns aren't allowed in methods without bodies
   --> /home/gtz/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-
serialize-0.3.19/src/serialize.rs:147:45
    |
147 |                                             &f_name: &str,
    |                                             ^^^^^^^

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -4,5 +4,5 @@ version = "1.1.0"
 authors = ["Jakub Matraszek <jakub.matraszek@gmail.com>"]
 
 [dependencies]
-midir = "0.2.0"
-rustbox = "*"
+midir = "0.5.0"
+rustbox = "0.11.0"

--- a/src/interface/mod.rs
+++ b/src/interface/mod.rs
@@ -24,21 +24,21 @@ impl Interface {
     }
 
     pub fn read_state<'a, 'b>(&'a mut self, state: State<'static>) -> Result<State<'b>, Box<Error>> {
-        let midi_out = try!(MidiOutput::new("Arturia Microbrute"));
-        let out_port: u32 = try!(Interface::get_midi_out_port(&midi_out));
-        let mut conn_out = try!(midi_out.connect(out_port, "microbrute").map_err(|e| e.kind()));
+        let midi_out = MidiOutput::new("Arturia Microbrute")?;
+        let out_port: usize = Interface::get_midi_out_port(&midi_out)?;
+        let mut conn_out = midi_out.connect(out_port, "microbrute")?;
 
-        let midi_in = try!(MidiInput::new("Arturia Microbrute"));
-        let in_port: u32 = try!(Interface::get_midi_in_port(&midi_in));
-        let conn_in = try!(midi_in.connect(in_port, "microbrute", |_, message, state| {
+        let midi_in = MidiInput::new("Arturia Microbrute")?;
+        let in_port: usize = Interface::get_midi_in_port(&midi_in)?;
+        let conn_in = midi_in.connect(in_port, "microbrute", |_, message, state| {
             response_handler::handle_incoming_midi_message(state, message)
-        }, state).map_err(|e| e.kind()));
+        }, state)?;
 
-        try!(conn_out.send(&request_handler::start_communication_command()));
+        conn_out.send(&request_handler::start_communication_command())?;
         sleep(Duration::from_millis(100));
         self.counter = 0;
         for cmd in request_handler::init_data() {
-            try!(conn_out.send(&request_handler::init_command(self.counter, cmd)));
+            conn_out.send(&request_handler::init_command(self.counter, cmd))?;
             sleep(Duration::from_millis(100));
             self.counter += 1;
         }
@@ -47,18 +47,18 @@ impl Interface {
     }
 
     pub fn set_state<'a, 'b>(&'a mut self, state: &'a mut State<'b>, param: &'b str, value: &'b str) -> Result<&State<'b>, Box<Error>> {
-        let midi_out = try!(MidiOutput::new("Arturia Microbrute"));
-        let out_port: u32 = try!(Interface::get_midi_out_port(&midi_out));
-        let mut conn_out = try!(midi_out.connect(out_port, "microbrute").map_err(|e| e.kind()));
-        try!(conn_out.send(&request_handler::set_command(self.counter, param, value)));
+        let midi_out = MidiOutput::new("Arturia Microbrute")?;
+        let out_port: usize = Interface::get_midi_out_port(&midi_out)?;
+        let mut conn_out = midi_out.connect(out_port, "microbrute")?;
+        conn_out.send(&request_handler::set_command(self.counter, param, value))?;
         sleep(Duration::from_millis(100));
         state.set(param, value);
         self.counter += 1;
         Ok(state)
     }
 
-    fn get_midi_out_port(midi_out: &MidiOutput) -> Result<u32, NotConnected> {
-        let mut out_port: Option<u32> = None;
+    fn get_midi_out_port(midi_out: &MidiOutput) -> Result<usize, NotConnected> {
+        let mut out_port: Option<usize> = None;
 
         for i in 0..midi_out.port_count() {
             match midi_out.port_name(i).unwrap().starts_with("MicroBrute") {
@@ -76,8 +76,8 @@ impl Interface {
         }
     }
 
-    fn get_midi_in_port(midi_in: &MidiInput) -> Result<u32, NotConnected> {
-        let mut in_port: Option<u32> = None;
+    fn get_midi_in_port(midi_in: &MidiInput) -> Result<usize, NotConnected> {
+        let mut in_port: Option<usize> = None;
 
         for i in 0..midi_in.port_count() {
             match midi_in.port_name(i).unwrap().starts_with("MicroBrute") {

--- a/src/main.rs
+++ b/src/main.rs
@@ -42,7 +42,7 @@ fn main() {
 fn run(rustbox: &RustBox) -> Result<(), Box<Error>> {
     let mut state = State::new();
     let mut midi_interface = Interface::new();
-    state = try!(midi_interface.read_state(state));
+    state = midi_interface.read_state(state)?;
 
     ui::print_state(&rustbox, &state);
     rustbox.present();
@@ -51,20 +51,20 @@ fn run(rustbox: &RustBox) -> Result<(), Box<Error>> {
             Ok(rustbox::Event::KeyEvent(key)) => {
                 match key {
                     Key::Esc => { break; } // Quit
-                    Key::Char('n') => { try!(ui::change_note_priority(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('v') => { try!(ui::change_velocity_response(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('p') => { try!(ui::change_play(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('q') => { try!(ui::change_seq_retrig(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('x') => { try!(ui::change_next_seq(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('o') => { try!(ui::change_step_on(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('t') => { try!(ui::change_step(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('l') => { try!(ui::change_lfo_key_retrig(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('e') => { try!(ui::change_env_legato_mode(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('g') => { try!(ui::change_gate(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('y') => { try!(ui::change_sync(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('b') => { try!(ui::change_bend_range(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('r') => { try!(ui::change_midi_recv_chan(&rustbox, &mut midi_interface, &mut state)); },
-                    Key::Char('s') => { try!(ui::change_midi_send_chan(&rustbox, &mut midi_interface, &mut state)); },
+                    Key::Char('n') => { ui::change_note_priority(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('v') => { ui::change_velocity_response(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('p') => { ui::change_play(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('q') => { ui::change_seq_retrig(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('x') => { ui::change_next_seq(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('o') => { ui::change_step_on(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('t') => { ui::change_step(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('l') => { ui::change_lfo_key_retrig(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('e') => { ui::change_env_legato_mode(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('g') => { ui::change_gate(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('y') => { ui::change_sync(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('b') => { ui::change_bend_range(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('r') => { ui::change_midi_recv_chan(&rustbox, &mut midi_interface, &mut state)?; },
+                    Key::Char('s') => { ui::change_midi_send_chan(&rustbox, &mut midi_interface, &mut state)?; },
                     _ => { }
                 }
                 ui::print_state(&rustbox, &state);

--- a/src/ui/mod.rs
+++ b/src/ui/mod.rs
@@ -51,9 +51,9 @@ pub fn change_note_priority<'a>(rustbox: &RustBox, midi_interface: &'a mut Inter
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "note_priority", "Low")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "note_priority", "Last")); },
-                Key::Char('3') => { try!(midi_interface.set_state(state, "note_priority", "High")); },
+                Key::Char('1') => { midi_interface.set_state(state, "note_priority", "Low")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "note_priority", "Last")?; },
+                Key::Char('3') => { midi_interface.set_state(state, "note_priority", "High")?; },
                 _ => { }
             }
         },
@@ -69,9 +69,9 @@ pub fn change_velocity_response<'a>(rustbox: &RustBox, midi_interface: &'a mut I
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "velocity_response", "/")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "velocity_response", "(")); },
-                Key::Char('3') => { try!(midi_interface.set_state(state, "velocity_response", ")")); },
+                Key::Char('1') => { midi_interface.set_state(state, "velocity_response", "/")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "velocity_response", "(")?; },
+                Key::Char('3') => { midi_interface.set_state(state, "velocity_response", ")")?; },
                 _ => { }
             }
         },
@@ -87,8 +87,8 @@ pub fn change_play<'a>(rustbox: &RustBox, midi_interface: &'a mut Interface, sta
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "play", "Note on")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "play", "Hold")); },
+                Key::Char('1') => { midi_interface.set_state(state, "play", "Note on")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "play", "Hold")?; },
                 _ => { }
             }
         },
@@ -104,9 +104,9 @@ pub fn change_seq_retrig<'a>(rustbox: &RustBox, midi_interface: &'a mut Interfac
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "seq_retrig", "Reset")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "seq_retrig", "Legato")); },
-                Key::Char('3') => { try!(midi_interface.set_state(state, "seq_retrig", "None")); },
+                Key::Char('1') => { midi_interface.set_state(state, "seq_retrig", "Reset")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "seq_retrig", "Legato")?; },
+                Key::Char('3') => { midi_interface.set_state(state, "seq_retrig", "None")?; },
                 _ => { }
             }
         },
@@ -122,9 +122,9 @@ pub fn change_next_seq<'a>(rustbox: &RustBox, midi_interface: &'a mut Interface,
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "next_seq", "End")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "next_seq", "Instant reset")); },
-                Key::Char('3') => { try!(midi_interface.set_state(state, "next_seq", "Instant continue")); },
+                Key::Char('1') => { midi_interface.set_state(state, "next_seq", "End")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "next_seq", "Instant reset")?; },
+                Key::Char('3') => { midi_interface.set_state(state, "next_seq", "Instant continue")?; },
                 _ => { }
             }
         },
@@ -140,8 +140,8 @@ pub fn change_step_on<'a>(rustbox: &RustBox, midi_interface: &'a mut Interface, 
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "step_on", "Clk")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "step_on", "Gate")); },
+                Key::Char('1') => { midi_interface.set_state(state, "step_on", "Clk")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "step_on", "Gate")?; },
                 _ => { }
             }
         },
@@ -157,10 +157,10 @@ pub fn change_step<'a>(rustbox: &RustBox, midi_interface: &'a mut Interface, sta
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "step", "1/4")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "step", "1/8")); },
-                Key::Char('3') => { try!(midi_interface.set_state(state, "step", "1/16")); },
-                Key::Char('4') => { try!(midi_interface.set_state(state, "step", "1/32")); },
+                Key::Char('1') => { midi_interface.set_state(state, "step", "1/4")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "step", "1/8")?; },
+                Key::Char('3') => { midi_interface.set_state(state, "step", "1/16")?; },
+                Key::Char('4') => { midi_interface.set_state(state, "step", "1/32")?; },
                 _ => { }
             }
         },
@@ -176,8 +176,8 @@ pub fn change_lfo_key_retrig<'a>(rustbox: &RustBox, midi_interface: &'a mut Inte
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "lfo_key_retrig", "On")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "lfo_key_retrig", "Off")); },
+                Key::Char('1') => { midi_interface.set_state(state, "lfo_key_retrig", "On")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "lfo_key_retrig", "Off")?; },
                 _ => { }
             }
         },
@@ -193,8 +193,8 @@ pub fn change_env_legato_mode<'a>(rustbox: &RustBox, midi_interface: &'a mut Int
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "env_legato_mode", "On")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "env_legato_mode", "Off")); },
+                Key::Char('1') => { midi_interface.set_state(state, "env_legato_mode", "On")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "env_legato_mode", "Off")?; },
                 _ => { }
             }
         },
@@ -210,9 +210,9 @@ pub fn change_gate<'a>(rustbox: &RustBox, midi_interface: &'a mut Interface, sta
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "gate", "Short")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "gate", "Medium")); },
-                Key::Char('3') => { try!(midi_interface.set_state(state, "gate", "Long")); },
+                Key::Char('1') => { midi_interface.set_state(state, "gate", "Short")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "gate", "Medium")?; },
+                Key::Char('3') => { midi_interface.set_state(state, "gate", "Long")?; },
                 _ => { }
             }
         },
@@ -228,9 +228,9 @@ pub fn change_sync<'a>(rustbox: &RustBox, midi_interface: &'a mut Interface, sta
     match rustbox.poll_event(false) {
         Ok(rustbox::Event::KeyEvent(key)) => {
             match key {
-                Key::Char('1') => { try!(midi_interface.set_state(state, "sync", "External")); },
-                Key::Char('2') => { try!(midi_interface.set_state(state, "sync", "Internal")); },
-                Key::Char('3') => { try!(midi_interface.set_state(state, "sync", "Auto")); },
+                Key::Char('1') => { midi_interface.set_state(state, "sync", "External")?; },
+                Key::Char('2') => { midi_interface.set_state(state, "sync", "Internal")?; },
+                Key::Char('3') => { midi_interface.set_state(state, "sync", "Auto")?; },
                 _ => { }
             }
         },
@@ -263,7 +263,7 @@ pub fn change_bend_range(rustbox: &RustBox, midi_interface: &mut Interface, stat
                         helpers::print_select_value_message(rustbox, "Bend range", "Select values using up and down arrows and press Enter to confirm.");
                     },
                     Key::Enter => {
-                        try!(midi_interface.set_state(state, "bend_range", BEND_RANGES[index as usize]));
+                        midi_interface.set_state(state, "bend_range", BEND_RANGES[index as usize])?;
                         break;
                     },
                     Key::Esc => {
@@ -304,7 +304,7 @@ pub fn change_midi_recv_chan(rustbox: &RustBox, midi_interface: &mut Interface, 
                         helpers::print_select_value_message(rustbox, "MIDI receive channel", "Select values using up and down arrows and press Enter to confirm.");
                     },
                     Key::Enter => {
-                        try!(midi_interface.set_state(state, "midi_recv_chan", MIDI_RECV_CHANNELS[index as usize]));
+                        midi_interface.set_state(state, "midi_recv_chan", MIDI_RECV_CHANNELS[index as usize])?;
                         break;
                     },
                     Key::Esc => {
@@ -345,7 +345,7 @@ pub fn change_midi_send_chan(rustbox: &RustBox, midi_interface: &mut Interface, 
                         helpers::print_select_value_message(rustbox, "MIDI send channel", "Select values using up and down arrows and press Enter to confirm.");
                     },
                     Key::Enter => {
-                        try!(midi_interface.set_state(state, "midi_send_chan", MIDI_SEND_CHANNELS[index]));
+                        midi_interface.set_state(state, "midi_send_chan", MIDI_SEND_CHANNELS[index])?;
                         break;
                     },
                     Key::Esc => {

